### PR TITLE
Improved concatenation way in PropertiesStorage.java

### PR DESCRIPTION
**Pull request by nikam14**
### Description

This PR... changed the '+' operator for concatenation to .concat method for concatenation because '+' operator takes more time then .concat method.
[Optimisation].

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

File: agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java [hunk 1]
```diff
@@ -92,11 +92,11 @@ public synchronized boolean configure(String name, Map<String, Object> params) {
             file = new File(path);
             try {
                 if (!file.createNewFile()) {
-                    s_logger.error("Unable to create _file: " + file.getAbsolutePath());
+                    s_logger.error("Unable to create _file: ".concat(file.getAbsolutePath()));
```
Comment (rohityadavcloud) [inline agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java:95]: the preferred way would be to use String.format, the concatenation using `+` isn't any different from concat.

**Review by DaanHoogland — APPROVED** [review]
clgtm, but why these particular instances of the + operator, @nikam14 ? Are these particular detrimental to performance in your system as opposed to the rest of the logging happening?

File: agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java (hunk 1 above)
Comment (nikam14) [reply agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java:95]: thank you for your suggestion , I will change it .

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Review by soreana — APPROVED** [review]
LGTM

**Comment by soreana** [comment]
@blueorangutan package

**Comment by soreana** [comment]
@blueorangutan package

**Comment by soreana** [comment]
@blueorangutan package

File: agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java [hunk 2]
```diff
@@ -92,11 +92,11 @@ public synchronized boolean configure(String name, Map<String, Object> params) {
             file = new File(path);
             try {
                 if (!file.createNewFile()) {
-                    s_logger.error("Unable to create _file: " + file.getAbsolutePath());
+                    s_logger.error(String.format("Unable to create _file: %s",file.getAbsolutePath()));
```
Comment (GutoVeronezi) [inline agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java:95]:
```suggestion
                    s_logger.error(String.format("Unable to create _file: %s", file.getAbsolutePath()));
```

File: agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java [hunk 3]
```diff
@@ -92,11 +92,11 @@ public synchronized boolean configure(String name, Map<String, Object> params) {
...
                     return false;
                 }
             } catch (IOException e) {
-                s_logger.error("Unable to create _file: " + file.getAbsolutePath(), e);
+                s_logger.error(String.format("Unable to create _file: %s",file.getAbsolutePath()), e);
```
Comment (GutoVeronezi) [inline agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java:99]:
```suggestion
                s_logger.error(String.format("Unable to create _file: %s", file.getAbsolutePath()), e);
```

**Comment by soreana** [comment]
@nikam14 Can you please address @GutoVeronezi 's comments?

**Comment by shwstppr** [comment]
@nikam14 can you please address outstanding review comments?

**Comment by DaanHoogland** [comment]
@nikam14 @GutoVeronezi , I've applied the rather trivial changes to move this forward, please publicaly shame me here or on email if that was just to brutal of me.

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java [hunk 4]
```diff
@@ -92,11 +92,11 @@ public synchronized boolean configure(String name, Map<String, Object> params) {
             file = new File(path);
             try {
                 if (!file.createNewFile()) {
-                    s_logger.error("Unable to create _file: " + file.getAbsolutePath());
+                    s_logger.error(String.format("Unable to create _file: %s", file.getAbsolutePath()));
                     return false;
                 }
             } catch (IOException e) {
-                s_logger.error("Unable to create _file: " + file.getAbsolutePath(), e);
+                s_logger.error(String.format("Unable to create _file: %s", file.getAbsolutePath()), e);
```
Comment (DaanHoogland) [inline agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java:99]:
```suggestion
                s_logger.error(String.format("Unable to create file: %s", file.getAbsolutePath()));
                if (s_logger.isDebugEnabled()) {
                    s_logger.bedug(String.format("IOException while trying to create file: %s", file.getAbsolutePath()), e);
                }
```

**Review by DaanHoogland — APPROVED** [review]
small improvement allowing operators to supress stacktraces

File: agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java (hunk 4 above)
Comment (DaanHoogland) [reply agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java:99]: can you concur @nikam14 ?

File: agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java [hunk 5]
```diff
@@ -92,11 +92,14 @@ public synchronized boolean configure(String name, Map<String, Object> params) {
...
                 if (!file.createNewFile()) {
-                    s_logger.error("Unable to create _file: " + file.getAbsolutePath());
+                    s_logger.error(String.format("Unable to create _file: %s", file.getAbsolutePath()));
                     return false;
                 }
             } catch (IOException e) {
-                s_logger.error("Unable to create _file: " + file.getAbsolutePath(), e);
+                s_logger.error(String.format("Unable to create file: %s", file.getAbsolutePath()));
+                if (s_logger.isDebugEnabled()) {
+                    s_logger.bedug(String.format("IOException while trying to create file: %s", file.getAbsolutePath()), e);
```
Comment (DaanHoogland) [inline agent/src/main/java/com/cloud/agent/dao/impl/PropertiesStorage.java:101]:
```suggestion
                    s_logger.debug(String.format("IOException while trying to create file: %s", file.getAbsolutePath()), e);
```

**Review by soreana — APPROVED** [review]

**Pull request merged**